cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -81,6 +81,24 @@ pub struct Registry {
     _json: Value,
 }
 
+#[derive(Debug, Clone)]
+pub enum LockResult {
+    /// Locking produced a new Lockfile.
+    /// The change could be a minimal as whitespace.
+    Changed(Lockfile),
```

Comment:
From the original comment i believe we might want to track the _old_ lockfile here as well.

Reply:
I'd prefer to defer that until we decide to combine `EditResult` and `UpgradeResult` which are the only cases where we need to expose it.